### PR TITLE
[BE] Remove redundant tests from test_mps for glu, sigmoid and tanh

### PR DESCRIPTION
**Pull request by Isalia20**
Remove redundant tests from test_mps.py for `glu`, `sigmoid` and `tanh` which are already tested by OpInfo tests:
- `test_glu`: `nn.functional.glu` OpInfo
- `test_sigmoid`: `sigmoid` OpInfo
- `test_tanh`: `tanh` OpInfo

You can run the tests with:
```bash
python test/test_mps.py -v \
  TestConsistencyMPS.test_output_match_nn_functional_glu_mps_float32 \
  TestConsistencyMPS.test_output_grad_match_nn_functional_glu_mps_float32 \
  TestConsistencyMPS.test_output_match_sigmoid_mps_float32 \
  TestConsistencyMPS.test_output_grad_match_sigmoid_mps_float32 \
  TestConsistencyMPS.test_output_match_tanh_mps_float32 \
  TestConsistencyMPS.test_output_grad_match_tanh_mps_float32
```

To confirm that they actually exist.

P.S. Of course these are not the only ops that need to be yanked out of test_mps.py but this is a start. I'll try to batch them in batches of 3 so it's more easily reviewable

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM, though for more involved test(especially ones that were added as regression fixes in the past) it would be nice to have links to OpInfo that mentions that the same shapes or dimensions are already covered

For example Claude produced following list:
> Deleted test: test_glu
> Deleted coverage: [4]/d0; (2,4)/d0,d1; (2,8,4,6)/d0–d3; fp32; fwd+bwd, random grad
> OpInfo samples on MPS: 46 samples, rank 1→6: (2,)d0, (8,2)d0/d1, (2,4)d0/d1, (8,2,4)d0–d2, (2,3,2,4)d0/d2/d3, (8,8,8)d0–d2, (2,3,8,8)d0/d2/d3, (8,3,6,8), (2,3,1,4,6,7), plus zero-size batch (0,2,4) etc.
> Verdict: Strict superset. Every deleted (shape, dim) pair has an equivalent-or-larger analogue. Also gains int32/uint8/bool/int8/int16 via dtypesIfMPS and fp16 grad.
> ────────────────────────────────────────
> Deleted test: test_sigmoid
> Deleted coverage: (2,3,4,5), (2,3,4), (2,8,4,5); fp32; fwd+bwd, grad=ones
>  OpInfo samples on MPS: 3 samples: (20,), (1,0,3) (empty), () (scalar)
> Verdict: Shape coverage narrows — no non-empty rank>1 sample. Gains: all MPS_DTYPES for test_output_match, fp32+fp16 for grad, and a random grad_output instead of ones.
> ────────────────────────────────────────
> Deleted test: test_tanh
> Deleted coverage: same as above
> OpInfo samples on MPS: 1 sample: (20, 20)
> Verdict: Same narrowing. Only one sample because supports_sparse_csr=True sends it down the 2-D branch of sample_inputs_elementwise_unary (torch/testing/_internal/opinfo/core.py:2611-2627) instead of the 1-D/empty/scalar branch.

**Comment by Isalia20** [comment]
yeah that's my plan. I started removing the ones initially which directly have counterparts on OpInfo tests, i.e. for example something being tested in test_mps.py with a shape of (6, ) and something being tested in OpInfo with a shape of (5,). (or if it's completely similar shapes then even better)

**Comment by Isalia20** [comment]
> LGTM, though for more involved test(especially ones that were added as regression fixes in the past) it would be nice to have links to OpInfo that mentions that the same shapes or dimensions are already covered
> 
> For example Claude produced following list:
> 
> > Deleted test: test_glu
> > Deleted coverage: [4]/d0; (2,4)/d0,d1; (2,8,4,6)/d0–d3; fp32; fwd+bwd, random grad
> > OpInfo samples on MPS: 46 samples, rank 1→6: (2,)d0, (8,2)d0/d1, (2,4)d0/d1, (8,2,4)d0–d2, (2,3,2,4)d0/d2/d3, (8,8,8)d0–d2, (2,3,8,8)d0/d2/d3, (8,3,6,8), (2,3,1,4,6,7), plus zero-size batch (0,2,4) etc.
> > Verdict: Strict superset. Every deleted (shape, dim) pair has an equivalent-or-larger analogue. Also gains int32/uint8/bool/int8/int16 via dtypesIfMPS and fp16 grad.
> > ────────────────────────────────────────
> > Deleted test: test_sigmoid
> > Deleted coverage: (2,3,4,5), (2,3,4), (2,8,4,5); fp32; fwd+bwd, grad=ones
> > OpInfo samples on MPS: 3 samples: (20,), (1,0,3) (empty), () (scalar)
> > Verdict: Shape coverage narrows — no non-empty rank>1 sample. Gains: all MPS_DTYPES for test_output_match, fp32+fp16 for grad, and a random grad_output instead of ones.
> > ────────────────────────────────────────
> > Deleted test: test_tanh
> > Deleted coverage: same as above
> > OpInfo samples on MPS: 1 sample: (20, 20)
> > Verdict: Same narrowing. Only one sample because supports_sparse_csr=True sends it down the 2-D branch of sample_inputs_elementwise_unary (torch/testing/_internal/opinfo/core.py:2611-2627) instead of the 1-D/empty/scalar branch.


I see, I'll take a look at the exact shapes if they have regression tests for specific shapes

**Comment by malfet** [comment]
Examples above are not blockers, because I think there are sufficient regression tests for single argument iterator ops, so that those shapes were never needed in the first place (or may be they were for easy MPSGraph days when some op shapes were hardcoded as 4D tensors at most), but such summary in followup PRs would be great, though obvious lines about dtypes and forward/backward is not that important.
